packages/pigeon/example/app/macos/Runner/messages.g.m

```diff
@@ -219,10 +219,32 @@ - (void)flutterMethodAString:(nullable NSString *)arg_aString
           @"dev.flutter.pigeon.pigeon_example_package.MessageFlutterApi.flutterMethod"
              binaryMessenger:self.binaryMessenger
                        codec:PGNMessageFlutterApiGetCodec()];
-  [channel sendMessage:@[ arg_aString ?: [NSNull null] ]
-                 reply:^(id reply) {
-                   NSString *output = reply;
-                   completion(output, nil);
-                 }];
+  [channel
+      sendMessage:@[ arg_aString ?: [NSNull null] ]
+            reply:^(NSArray<id> *reply) {
+              if (reply != nil) {
+                if (reply.count > 1) {
+                  completion(nil, [FlutterError errorWithCode:reply[0]
+                                                      message:reply[1]
+                                                      details:reply[2]]);
+                } else if (reply[0] == nil) {
+                  completion(
+                      nil,
+                      [FlutterError
+                          errorWithCode:@"null-error"
+                                message:
+                                    @"Flutter api returned null value for non-null return value."
```

Comment:
This case can't happen without a bug in our Dart generator (with any reasonably current version of Dart), so we don't need runtime handling for it.